src/engine/Service.cpp

```diff
@@ -0,0 +1,175 @@
+      HttpClient client(serviceUrl.host(), serviceUrl.port());
+      return client.sendRequest(boost::beast::http::verb::post,
+                                serviceUrl.host(), serviceUrl.target(),
+                                serviceQuery, "application/sparql-query",
+                                "text/tab-separated-values");
+    } else {
+      HttpsClient client(serviceUrl.host(), serviceUrl.port());
+      return client.sendRequest(boost::beast::http::verb::post,
+                                serviceUrl.host(), serviceUrl.target(),
+                                serviceQuery, "application/sparql-query",
+                                "text/tab-separated-values");
+    }
+  }();
+  LOG(DEBUG) << "TSV result is:" << std::endl << tsvResult.str() << std::endl;
+
+  // Turn TSV into VALUES clause, which will then be turned into a `ResultTable`
+  // by `parsedQuery::Values` (namely, method `writeValues`, which also
+  // constructs a `localVocab` if required, which it usually is for SERVICE
+  // queries).
+  //
+  // TODO: Is the time needed for turning the `parsedQuery::Values` into a
+  // `ResultTable` accounted for?
+  //
+  // TODO: Eventually, we should construct a `ResultTable` directly, with a mix
+  // of IDs from this endpoint and IDs from the remote endpoint (the latter with
+  // a bitmask, to that we can easily distinguish them).
+  parsedQuery::SparqlValues sparqlValues;
+  std::string variables;
+  if (!std::getline(tsvResult, variables)) {
+    throw std::runtime_error(absl::StrCat("Response from SPARQL endpoint ",
+                                          serviceUrl.host(), " is empty"));
+  }
+  LOG(INFO) << "First line of response body: " << variables << std::endl;
+  for (const std::string_view& variableName : absl::StrSplit(variables, "\t")) {
+    sparqlValues._variables.push_back(Variable{std::string{variableName}});
+  }
+  // TODO: This duplicates code in SparqlQleverVisitor.cpp . Maybe it would be
+  // more meaningful to postpone the interpretation of the values from a VALUES
+  // clause (via `parseTripleObject`) to parsedQuery::Values ?
+  std::string line;
+  std::string lastLine;
+  while (lastLine = std::move(line), std::getline(tsvResult, line)) {
+    if (sparqlValues._values.size() == 0) {
+      LOG(INFO) << "Second line of response body: " << line << std::endl;
+    }
+    std::vector<TripleComponent> valueTuple;
+    for (auto value : absl::StrSplit(line, "\t")) {
+      valueTuple.push_back(
+          TurtleStringParser<TokenizerCtre>::parseTripleObject(value));
+    }
+    sparqlValues._values.push_back(valueTuple);
+  }
+  if (sparqlValues._values.size() > 1) {
+    LOG(INFO) << "Last line of response body: " << lastLine << std::endl;
+  }
+  resultAsValues_ = std::make_unique<Values>(qec, std::move(sparqlValues));
+}
+
+// ____________________________________________________________________________
+string Service::asStringImpl(size_t indent) const {
```

Comment:
Maybe (consistently also in the header)
write `std::string` and `std::vector`, we do this also in large parts of the rest of the codebase.
At some point in the near future we should eliminate the remaining `using std::string` in widely included headers.